src/lib/network_pool/transaction_pool.ml

```diff
@@ -541,6 +589,8 @@ struct
                        Transaction_hash.User_command_with_valid_signature
                        .to_yojson locally_generated_dropped ) )
             ] ;
+      List.iter locally_generated_dropped
```

Comment:
Shouldn't you be iterating through all dropped transactions, not just locally dropped ones? You are missing some dropped transactions here.